test/recipes/02-test_windows_registry.t

```diff
@@ -0,0 +1,83 @@
+#! /usr/bin/env perl
+# Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+use OpenSSL::Test;
+use OpenSSL::Test::Utils;
+use OpenSSL::Test qw/:DEFAULT bldtop_file data_file srctop_file cmdstr/;
+
+setup("test_windows_registry");
+
+plan skip_all => "Windows registry tests are only available on windows"
+    if $^O !~ /(MSWin)/;
+
+my $actual;
+my $expect;
+
+my @tempout = run(app(["openssl", "version", "-w"]), capture => 1);
+my $context = "@tempout";
+$context =~ s/^.*: //;
+
+
+@tempout = run(app(["openssl", "version", "-v"]), capture => 1);
+my $version = "@tempout";
+$version =~ s/^OpenSSL //;
+$version =~ s/\.[0-9]\+-.*\n//;
+
+my $regkey = "HKLM\\SOFTWARE\\OpenSSL-".$version."-".$context;
+$regkey =~ s/\n//g;
+print "REGKEY IS $regkey\n";
+
+my $exit = run(cmd(["reg.exe", "query", $regkey, "/v", "OPENSSLDIR", "/reg:32"]));
+
+plan skip_all => "Skipping test as registry keys aren't set"
+    if $exit == 0;
+
+plan tests => 3;
+
+my @expectossldir = run(cmd(["reg.exe", "query", $regkey, "/reg:32", "/t", "REG_EXPAND_SZ", "/v", "OPENSSLDIR"]), capture => 1);
+
+my @expectengdir = run(cmd(["reg.exe", "query", $regkey, "/reg:32", "/t", "REG_EXPAND_SZ", "/v", "ENGINESDIR"]), capture => 1);
+
+my @expectmoddir = run(cmd(["reg.exe", "query", $regkey, "/reg:32", "/t", "REG_EXPAND_SZ", "/v", "MODULESDIR"]), capture => 1);
+
+my @osslversion = run(app(["openssl", "version", "-d"]), capture => 1);
```

Comment:
Seems strange to call this osslversion. I might have expected ossldir or something.